cmd/immuadmin/command/hot_backup.go

```diff
@@ -0,0 +1,661 @@
+	binary.BigEndian.PutUint32(payload[txSignSizeOffset:], uint32(len(signature)))
+	binary.BigEndian.PutUint32(payload[txSizeOffset:], uint32(len(content)))
+	payload = append(payload, signature...)
+	payload = append(payload, content...)
+
+	_, err := output.Write(payload)
+	return err
+}
+
+type restoreParams struct {
+	input    string
+	append   bool
+	progress bool
+	force    bool
+	verify   bool
+}
+
+func (cl *commandlineHotBck) hotRestore(cmd *cobra.Command) {
+	ccmd := &cobra.Command{
+		Use:   "hot-restore <db_name>",
+		Short: "Restore data from backup file",
+		Long: "Restore saved transaction from backup file without stopping the database engine. " +
+			"Restore can restore the data from scratch or apply only the missing data.",
+		PersistentPreRunE: cl.ConfigChain(cl.connect),
+		PersistentPostRun: cl.disconnect,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			params, err := prepareRestoreParams(cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			file := io.Reader(os.Stdin)
+			if params.input != "-" {
+				f, err := os.Open(params.input)
+				if err != nil {
+					return err
+				}
+				file = f
+				defer f.Close()
+			}
+
+			if params.verify {
+				return verifyFile(file)
+			}
+
+			dbExist, err := cl.isDbExists(args[0])
+			if err != nil {
+				return err
+			}
+
+			var signature, payload []byte
+			var firstTx uint64
+			if dbExist {
+				firstTx, signature, payload, err = cl.verifyDb(params, args[0], file)
+				if err != nil {
+					return err
+				}
+				err = cl.immuClient.UpdateDatabase(cl.context, &schema.DatabaseSettings{DatabaseName: args[0], Replica: true})
+				if err != nil {
+					return fmt.Errorf("cannot switch on replica mode for db: %v", err)
+				}
+			} else {
+				// db does not exist - create as replica and use it
+				err = cl.immuClient.CreateDatabase(cl.context, &schema.DatabaseSettings{DatabaseName: args[0], Replica: true, MasterDatabase: "dummy"})
+				if err != nil {
+					return err
+				}
+
+				udr, err := cl.immuClient.UseDatabase(cl.context, &schema.Database{DatabaseName: args[0]})
+				if err != nil {
+					return err
+				}
+				cl.context = metadata.NewOutgoingContext(cl.context, metadata.Pairs("authorization", udr.GetToken()))
+			}
+			defer func() {
+				err = cl.immuClient.UpdateDatabase(cl.context, &schema.DatabaseSettings{DatabaseName: args[0], Replica: false})
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error switching off replica mode for db: %v", err)
+				}
+			}()
+
+			return cl.runHotRestore(file, params.progress, firstTx, signature, payload)
+		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			verify, _ := cmd.Flags().GetBool("verify")
+			if verify {
+				return cobra.ExactArgs(0)(cmd, args)
+			}
+			return cobra.ExactArgs(1)(cmd, args)
+		},
+	}
+	ccmd.Flags().StringP("input", "i", "-", "input file file, \"-\" for stdin")
+	ccmd.Flags().Bool("verify", false, "do not restore data, only verify backup")
```

Comment:
verify-only ?

Reply:
Agree, `verify-only` is more clear, will change

Reply:
changed